packages/fetchai/skills/generic_seller/behaviours.py

```diff
@@ -86,6 +91,29 @@ def teardown(self) -> None:
         self._unregister_service()
         self._unregister_agent()
 
+    def _retry_failed_registration(self) -> None:
+        """
+        Retry a failed registration.
+
+        :return: None
+        """
+        if self.failed_registration_msg is not None:
+            self._nb_retries += 1
+            if self._nb_retries >= self._max_retries:
+                self.context.is_active = False
+                return
+
+            oef_search_dialogues = cast(
+                OefSearchDialogues, self.context.oef_search_dialogues
+            )
+            oef_search_msg, _ = oef_search_dialogues.create(
+                counterparty=self.failed_registration_msg.to,
+                performative=self.failed_registration_msg.performative,
+                service_description=self.failed_registration_msg.service_description,
+            )
+            self.context.outbox.put_message(message=oef_search_msg)
+            self.context.logger.info("retrying registration on SOEF.")
```

Comment:
we already have this in `_register_agent` so call it, not repeat it.

Reply:
This code retires any failed registration, agent or services. `_register_agent` only registers the agent.